### PR TITLE
update to kraken2: breaking change - output channels renamed

### PR DESCRIPTION
**Pull request by lescai**
## PR checklist

Closes #XXX 

- [ ] The existing version of the module did not output a taxonomic classification of *each* input reads, which is needed for the pipeline hgtseq. by missing the command option --output this classification is output on stderr. After conversation on Slack, it was also realised that outputting classified fastq files would be best an optional output. The module has been therefore recoded by adding to additional channels, with boolean values to indicate if fastq classified/unclassified reads should be written and whether classification of each reads should be returned.
- [ ] Tests have been successfully run with singularity
- [ ] No new tools added
- [ ] Test data already online
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Review by jfy133 — COMMENTED** [review]
I see you've ot finished yet (still TODO strings etc), but overall is looking good

One bit of reedback, I find classified/unclassified/classifiedreads to be quite ambiguous what they are... maybe we can come up with a better naming scheme for both flags ?

What about for input channels:

```
save_output_fastqs
save_reads_report
```

And for variables:

```
classified_reads_fastq
unclassified_reads_fastq
classified_reads_report
```

or something like that? I think it'll be faster for a developer to parse that way.

**Comment by lescai** [comment]
Hi @jfy133 
completely agree. 
I've modified as 
`save_output_fastqs` and
`save_reads_assignment` 
(this because kraken has a default output called "report" so we don't mix the two).

as a consequence the outputs have been renamed to
`classified_reads_fastq`
`unclassified_reads_fastq`
`classified_reads_assignment`
`report`

both docker and singularity tests are passing here and on my HPC, except the usually annoying conda here.

File: modules/kraken2/kraken2/meta.yml [hunk 1]
```diff
@@ -27,25 +27,40 @@ input:
...
   - meta:
       type: map
       description: |
         Groovy Map containing sample information
         e.g. [ id:'test', single_end:false ]
-  - classified:
+  - classified_reads_fastq:
       type: file
       description: |
         Reads classified to belong to any of the taxa
```
Comment (jfy133) [inline modules/kraken2/kraken2/meta.yml:49]:
```suggestion
        Reads classified as belonging to any of the taxa
```

File: modules/kraken2/kraken2/meta.yml [hunk 2]
```diff
@@ -27,25 +27,40 @@ input:
...
         on the Kraken2 database.
       pattern: "*{fastq.gz}"
-  - unclassified:
+  - unclassified_reads_fastq:
       type: file
       description: |
         Reads not classified to belong to any of the taxa
```
Comment (jfy133) [inline modules/kraken2/kraken2/meta.yml:55]:
```suggestion
        Reads not classified to any of the taxa
```

File: modules/kraken2/kraken2/meta.yml [hunk 3]
```diff
@@ -27,25 +27,40 @@ input:
...
         on the Kraken2 database.
       pattern: "*{fastq.gz}"
-  - txt:
+  - classified_reads_assignment:
+      type: file
+      description: |
+        Kraken2 output file indicating the taxonomic assignment of
+        each input reads
```
Comment (jfy133) [inline modules/kraken2/kraken2/meta.yml:62]:
```suggestion
        each input read
```

**Review by jfy133 — APPROVED** [review]
A couple of minor phrasing changes in the meta.

Make sure to that the  commit message is very clear when squash/merge that this is a 'breaking' change/major update (or just that you're renaming the output channels)

**Pull request merged**